src/block/progress-circle/edit.js

```diff
@@ -0,0 +1,118 @@
+import ProgressCircleStyles from './style'
+import { InspectorTabs } from '~stackable/components'
+import {
+	BlockDiv,
+	Alignment,
+	Advanced,
+	Responsive,
+	Transform,
+	EffectsAnimations,
+	CustomAttributes,
+	CustomCSS,
+	ConditionalDisplay,
+	ProgressBar,
+	useGeneratedCss,
+	Typography,
+	getTypographyClasses,
+	getAlignmentClasses,
+} from '~stackable/block-components'
+import { version as VERSION } from 'stackable'
+import { withBlockAttributeContext, withQueryLoopContext } from '~stackable/higher-order'
+import classnames from 'classnames'
+import striptags from 'striptags'
+
+import { compose } from '@wordpress/compose'
+import { useState, useEffect } from '@wordpress/element'
+
+const Edit = ( {
+	className, attributes,
+} ) => {
+	const [ forceRedraw, setForceRedraw ] = useState( false )
+	useGeneratedCss( attributes )
+
+	const blockAlignmentClass = getAlignmentClasses( attributes )
+	const textClasses = getTypographyClasses( attributes )
+
+	const blockClassNames = classnames( [
+		className,
+		'stk-block-progress-circle',
+		blockAlignmentClass,
+	] )
+
+	const textClassNames = classnames( [
+		'stk-progress-circle__inner-text',
+		textClasses,
+	] )
+
+	const derivedValue = `${ attributes.textPrefix.trim() }${ attributes.progressPercent }${ attributes.textSuffix.trim() }`.trim()
+
+	/**
+	 * workaround to correct the circle width when size changes
+	 */
+	 const workAroundClass = classnames( {
+		'force-redraw': forceRedraw,
+	} )
+
+	useEffect( () => {
+		if ( attributes.progressSize ) {
+			setForceRedraw( true )
+			const timer = setTimeout( () => {
+				setForceRedraw( false )
+			}, 100 )
```

Comment:
The delay is a bit noticeable, I tried removing the delay (this defaults it to `0`) and it seems that this works.
```suggestion
			} )
```